### PR TITLE
Stricter asserts on _bn_to_int and _int_to_bn

### PR DESCRIPTION
**Pull request by public**
Helped me catch my terrible programming when working on DH

### CODE REVIEW

File: src/cryptography/hazmat/backends/openssl/backend.py [hunk 1]
```diff
@@ -346,6 +347,7 @@ def _int_to_bn(self, num, bn=None):
         ownership of the object). Be sure to register it for GC if it will
         be discarded after use.
         """
+        assert (bn is None) or (bn is not None and bn != self._ffi.NULL)
```
Comment (alex) [inline src/cryptography/hazmat/backends/openssl/backend.py]: I think this simplifies to `assert bn is None or bn != self._ffi.NULL`? The second `bn is not None` is a tautology.

Comment (public) [inline src/cryptography/hazmat/backends/openssl/backend.py]: `bn = None` means that `bn` is immediately set to `NULL`. However if a non-None value is provided then write the value to that instead.

This assert removes the ambiguity of the `_int_to_bn(0, NULL)` vs `_int_to_bn(0)`.

Comment (alex) [inline src/cryptography/hazmat/backends/openssl/backend.py]: Right, I don't think the line I proposed is semantically different from what you have, just a bit shorter.

Comment (public) [inline src/cryptography/hazmat/backends/openssl/backend.py]: You are correct :)

**Comment by public** [comment]
Updated.

**Comment by reaperhulk** [comment]
:+1: will merge once tests pass. C'mon travis.

**Pull request merged**